builder/builder.go

```diff
@@ -1311,6 +1312,102 @@ func (b *Builder) UpdateMixVer(version int) error {
 	return ioutil.WriteFile(filepath.Join(b.Config.Builder.VersionPath, b.MixVerFile), []byte(b.MixVer), 0644)
 }
 
+// UpdateOsRelease sets the mix version in the full-chroot os-release file
+func (b *Builder) UpdateOsRelease(version string) error {
+	filename := filepath.Join(b.Config.Builder.ServerStateDir, "image", b.MixVer, "full/usr/lib/os-release")
+
+	f, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = f.Close()
+	}()
+
+	// TODO: If this is a mix, NAME and ID should probably change too. Create a section in
+	// configuration that will be used as reference to fill this.
+	// TODO: If this is a mix, add extra field for keeping track of the Clear Linux version
+	// used. Maybe also put the UPSTREAM URL, so we are ready to support mixes of mixes.
+	//
+	// See also: https://github.com/clearlinux/mixer-tools/issues/113
+
+	var newBuf bytes.Buffer
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		text := scanner.Text()
+		if strings.HasPrefix(text, "VERSION_ID=") {
+			text = "VERSION_ID=" + version
+		}
+		_, err = fmt.Fprintln(&newBuf, text)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = scanner.Err()
+	if err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(filename, newBuf.Bytes(), 0644)
+}
+
+// UpdateFormatFile update the format number in the full-chroot format file
+func (b *Builder) UpdateFormatFile(version int) error {
+	formatFile := filepath.Join(b.Config.Builder.ServerStateDir, "image", b.MixVer, "full/usr/share/defaults/swupd/format")
+	if _, err := os.Stat(formatFile); err == nil {
+		return err
+	}
+
+	return ioutil.WriteFile(formatFile, []byte(strconv.Itoa(version)), 0644)
+}
+
+// ModifyBundles goes through the bundle directory and performs an action when it finds
+// a Deprecated bundle
+func (b *Builder) ModifyBundles(action func(string) error) error {
+	path := b.Config.Mixer.LocalBundleDir
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return err
+	}
+
+	var scanner *bufio.Scanner
+	for _, file := range files {
+		fmt.Println("CHECKING FILE: " + file.Name())
+		fileToScan := filepath.Join(path, file.Name())
+		f, err := os.Open(fileToScan)
+		if err != nil {
+			return err
+		}
+
+		// Scan the files and find which bundle definitions are marked deprecated
+		scanner = bufio.NewScanner(f)
+		var str string
+		re := regexp.MustCompile("#\\s\\[STATUS\\]:\\s*Deprecated.*")
```

Comment:
instead of using double quotes (") use back-tics (\`) so you don't have to escape your escapes.
This seems to work
```go
re := regexp.MustCompile(`#\s\[STATUS\]:\s*Deprecated.*`)
```
https://goplay.space/#hhjZcWAeWOW

Reply:
Also I like that link... `WAeWOW` like it's impressed with my code. :rofl: